java/server/src/org/openqa/selenium/grid/distributor/local/Host.java

```diff
@@ -160,25 +162,33 @@ public UUID getId() {
         nodeId,
         uri,
         getHostStatus() == UP,
+        node.isDraining(),
         maxSessionCount,
         stereotypes,
         used);
   }
 
-
   public Status getHostStatus() {
     return status;
   }
 
   /**
    * @return The previous status of the node.
    */
-  private Status setHostStatus(Status status) {
+  Status setHostStatus(Status status) {
     Status toReturn = this.status;
     this.status = Require.nonNull("Status", status);
+    if (status.equals(DRAINING) && !node.isDraining()) {
+      node.drain();
+    }
     return toReturn;
   }
 
+  /**
+   * Whether or not the number of sessions is less than maxSessions
```

Comment:
This adds nothing that's not already expressed by the method signature.

Reply:
Ok. I'm not offended by over-explaining things (I'm sure that's obvious by now), but I don't mind leaving it out.